### ISSUE TITLE
[BUG][DEV] Inconsistent ranged projectile velocity on small caliber projectiles

### ISSUE THREAD

**Issue by TwistedTail** [Stooberton/ACF-3#199]
**Short Description**

Using the ACF.GetRangedSpeed function to get the approximated velocity of a projectile after a certain range can give really inconsistent results with small caliber projectiles.

**Branch and Version**

Latest dev version.

**How to Reproduce (Optional)**

By maxing out the propellant slider on a 5.56mm machinegun ammo setup, we can easily make this formula give us nearly 5.7km/s expected velocity at 800m, much higher than the muzzle velocity.

**Screenshots (Optional)**

![hl2_XtHlmcprwm](https://user-images.githubusercontent.com/8784231/117379421-6d35a080-aea5-11eb-8091-cdb450f2dbe4.png)

**Comment by Alexandre425** [comment]
A quick and dirty iterative method might be a good solution, something along the lines of

```
proj_vel = muzzle_vel
for i, 8 iterations do
  time = 100m / proj_vel
  proj_vel = applyDrag(proj_vel, time)
  vel[i] = proj_vel
end

display values for i = 3 (300m) and i = 8 (800m)
```

**Comment by TwistedTail** [comment]
Fixed on bbc273e. I just inverted the formula to find out at which range the speed will be 0. If the given range is higher than this limit, the result will be defaulted to 0.
